arithmetization/src/test/java/net/consensys/linea/zktracer/module/hub/TxWarmTest.java

```diff
@@ -0,0 +1,224 @@
+
+  // sender account
+  private static final KeyPair senderKeyPair = new SECP256K1().generateKeyPair();
+  private static final Address senderAddress =
+      Address.extract(Hash.hash(senderKeyPair.getPublicKey().getEncodedBytes()));
+  private static final ToyAccount senderAccount =
+      ToyAccount.builder().balance(Wei.fromEth(123)).nonce(12).address(senderAddress).build();
+
+  // receiver account: 0+1 SMC account
+  private static final ToyAccount receiverAccount =
+      ToyAccount.builder()
+          .balance(Wei.fromEth(1))
+          .address(Address.fromHexString("0xdead000000000000000000000000000beef"))
+          .code(BytecodeCompiler.newProgram().push(1).push(0).op(OpCode.ADD).compile())
+          .build();
+
+  @Test
+  void inefficientAccessList() {
+    // inefficient access list
+    final List<AccessListEntry> accessList = new ArrayList<>(List.of());
+
+    // warming a PRC wo storage keys
+    accessList.add(new AccessListEntry(Address.BLAKE2B_F_COMPRESSION, NO_STORAGE_KEYS));
+
+    // warming a PRC with storage keys
+    accessList.add(
+        new AccessListEntry(Address.MODEXP, List.of(Bytes32.random(), Bytes32.random())));
+
+    // warming 0x00 Address
+    accessList.add(new AccessListEntry(OxO_ADDRESS, NO_STORAGE_KEYS));
+    accessList.add(new AccessListEntry(OxO_ADDRESS, List.of(Bytes32.ZERO, FIXED_STORAGE_KEYS)));
+
+    // warming dead beef address
+    accessList.add(new AccessListEntry(FIXED_ADDRESS, NO_STORAGE_KEYS));
+
+    // warming dead beef address a second time
+    accessList.add(new AccessListEntry(FIXED_ADDRESS, NO_STORAGE_KEYS));
+
+    // warming dead beef address a third time, with random storage keys
+    accessList.add(new AccessListEntry(FIXED_ADDRESS, List.of(Bytes32.random(), Bytes32.random())));
+
+    // warming dead beef address a fourth time, with defined storage keys
+    accessList.add(new AccessListEntry(FIXED_ADDRESS, List.of(FIXED_STORAGE_KEYS)));
+
+    // warming dead beef address a fifth time, with defined storage keys
+    accessList.add(
+        new AccessListEntry(
+            FIXED_ADDRESS,
+            List.of(
+                FIXED_STORAGE_KEYS,
+                Bytes32.random(),
+                Bytes32.ZERO,
+                Bytes32.random(),
+                FIXED_STORAGE_KEYS)));
+
+    final Transaction tx =
+        ToyTransaction.builder()
+            .sender(senderAccount)
+            .to(receiverAccount)
+            .keyPair(senderKeyPair)
+            .gasLimit(300000L)
+            .transactionType(TransactionType.ACCESS_LIST)
+            .accessList(accessList)
+            .value(Wei.of(1000))
+            .build();
+
+    ToyExecutionEnvironmentV2.builder()
+        .accounts(List.of(senderAccount, receiverAccount))
+        .transaction(tx)
+        .zkTracerValidator(zkTracer -> {})
+        .build()
+        .run();
+  }
+
+  @Test
+  void warmSender() {
```

Comment:
These tests would benefit from being made parametric e.g. via `@MethodSource` to test more thoroughly.